File: M3Tools/Types/Orders/CartItem.vb
Namespace Types
	Public Class CartItem
		Private Property Product As Product
		Public Property Quantity As Integer

		ReadOnly Property ItemTotal As Double
			Get
				Return Me.Product.Price * Me.Quantity
			End Get
		End Property

		Public ReadOnly Property ItemName As String
			Get
				Return Product.Name
			End Get
		End Property

		Public ReadOnly Property ItemID As Integer
			Get
				Return Product.Id
			End Get
		End Property

		Public ReadOnly Property ItemPrice As Double
			Get
				Return Product.Price
			End Get
		End Property

		Public Sub New()
			Me.New(New Product(), 0)
		End Sub

		Public Sub New(product As Product, quantity As Integer)
			Me.Product = product
			Me.Quantity = quantity
		End Sub
	End Class
End Namespace

File: M3Tools/Types/Orders/CartItem.cs
namespace SPPBC.M3Tools.Types
{
	/// <summary>
	/// An item with in the cart.
	/// Contains item info as well as quantity
	/// </summary>
	public class CartItem
	{
		/// <summary>
		/// The item for this cart item entry
		/// </summary>
		public Product Item { get; set; }

		/// <summary>
		/// How many of this item to have in the cart
		/// </summary>
		public int Quantity { get; set; }

		/// <summary>
		/// 
		/// </summary>
		public string ItemName => Item.Name;

		/// <summary>
		/// The complete total for the amount of the item in the cart
		/// </summary>
		public double ItemTotal => (double)(Item.Price * Quantity);

		/// <summary>
		/// 
		/// </summary>
		/// <param name="product"></param>
		/// <param name="quantity"></param>
		public CartItem(Product product, int quantity)
		{
			Item = product;
			Quantity = quantity;
		}

		/// <summary>
		/// 
		/// </summary>
		/// <param name="left"></param>
		/// <param name="right"></param>
		/// <returns></returns>
		public static bool operator ==(CartItem left, CartItem right)
		{
			return !(left is null ^ right is null) && left?.Item == right?.Item;
		}

		/// <summary>
		/// 
		/// </summary>
		/// <param name="left"></param>
		/// <param name="right"></param>
		/// <returns></returns>
		public static bool operator !=(CartItem left, CartItem right)
		{
			return !(left == right);
		}
	}
}
